Extension/src/LanguageServer/client.ts

```diff
@@ -539,6 +539,8 @@ export interface ChatContextResult {
     compiler: string;
     targetPlatform: string;
     targetArchitecture: string;
+    compilerArgs?: string[];
```

Comment:
Yeah, I think we should only include flags that we know are likely to affect the AI generated code or responses and not flags that just impact the compiler details (optimizations, warnings, etc.). And maybe each flag could be A/B tested separately.

Reply:
Also are the "defines" going to be A/B tested separately so we can determine if that improves stuff or not?

Reply:
And see my comment on the cpptools side.

Reply:
Thanks. I changed the PR to better support compiler arguments A/B experiments, so we'll know better which arguments to pass to Copilot. Which arguments to pass is controlled by experimental flags.

For user defines, I think it makes more sense to just pass the ones that are referenced in current document instead of all, need to check how to do that on cpptools side.